GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1175,9 +1175,11 @@ private void UpdatePlaceholders()
                             // 1. not in the projection anymore
                             // 2. in the projection but is not a folder in the projection
                             // 3. Folder placeholder is a possible tombstone
-                            if (!isProjected ||
+                            // and if the folder is not the root folder of the repository
+                            if ((!isProjected ||
                                 !isFolder ||
-                                folderPlaceholder.IsPossibleTombstoneFolder)
+                                folderPlaceholder.IsPossibleTombstoneFolder) &&
+                                folderPlaceholder.Path != string.Empty)

```

Comment:
How about we add the string.Empty to the `folderPlaceholdersToKeep` when we initialize it since we will always be keeping that folder?

Reply:
I thought about that too, but was concerned it might "match against everything" and short-circuit all the path comparisons.  However, a little testing and a more careful read of the code shows this is not at all the case, so I rebased with your suggestion implemented in 46cca89.  Thanks!